src/blockchain/deposit_strategy/interface.py

```diff
@@ -1,17 +1,44 @@
+import logging
 from abc import ABC, abstractmethod
 
 from blockchain.typings import Web3
+from cryptography.verify_signature import compute_vs
+from transport.msg_types.deposit import DepositMessage
+from web3.contract.contract import ContractFunction
+
+logger = logging.getLogger(__name__)
 
 
 class ModuleDepositStrategyInterface(ABC):
+    BLOCKS_IN_ONE_DAY = 24 * 60 * 60 // 12
+
+    CACHE_BLOCK_AMOUNT = 300
+    REQUEST_SIZE = 1024
+
     def __init__(self, w3: Web3, module_id: int):
         self.w3 = w3
         self.module_id = module_id
+        self._gas_fees: list = []
 
-    @abstractmethod
-    def is_gas_price_ok(self) -> bool:
-        pass
+        # Used for caching
+        self._latest_fetched_block: int = 0
+        self._days_param: int = 0
 
     @abstractmethod
-    def is_deposited_keys_amount_ok(self) -> bool:
+    def prepare_and_send(self, quorum: list[DepositMessage], with_flashbots: bool) -> bool:
         pass
+
+    @staticmethod
+    def _prepare_signs_for_deposit(quorum: list[DepositMessage]) -> tuple[tuple[str, str], ...]:
```

Comment:
Maybe move this to another service that will know how to send and handle different deposit transactions?

Reply:
But this is a static method, do you want to move it to some class?

Reply:
Away from deposit strategy class

Reply:
Moved